memo/src/processor.rs

```diff
@@ -0,0 +1,67 @@
+use crate::{error::MemoError, instruction::Instruction};
+use solana_sdk::{
+    account_info::AccountInfo, entrypoint, entrypoint::ProgramResult,
+    program_error::PrintProgramError, pubkey::Pubkey,
+};
+
+entrypoint!(process_instruction);
+fn process_instruction<'a>(
+    program_id: &Pubkey,
+    accounts: &'a [AccountInfo<'a>],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    if let Err(error) = process(program_id, accounts, instruction_data) {
+        // catch the error so we can print it
+        error.print::<MemoError>();
```

Comment:
Printing seems unnecessary here to me.  The program is trivial and it returns nice custom error codes